portfolio/src/main/java/com/google/sps/servlets/ThumbsDownServlet.java

```diff
@@ -77,8 +115,56 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
         // Add the updated entity back in the datastore
         datastore.put(commentEntity);
+    }
 
-        response.sendRedirect("/contact.html");
-        return;
+    // Accesses the datastore to get the UserInfo entity. Returns the entity or null if one does not exist.
+    private Entity getUserInfoEntity() {
+        UserService userService = UserServiceFactory.getUserService();
+        String id = userService.getCurrentUser().getUserId();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Filter queryFilter = new FilterPredicate("id", Query.FilterOperator.EQUAL, id);
+        Query query = new Query("UserInfo").setFilter(queryFilter);
+        PreparedQuery results = datastore.prepare(query); 
```

Comment:
You can in-line these last three lines like so:

return datastore.prepare(query).asSingleEntity();

Reply:
Cool! I'll add this one too